utils/compare_minions.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+
+import salt.config
+import salt.loader
+
+import json
+import json_delta
+
+try:
+    from colorama import Fore, Back, Style, init
+    init()
+except ImportError:  # fallback so that the imported classes always exist
+    class ColorFallback():
+        __getattr__ = lambda self, name: ''
+    Fore = Back = Style = ColorFallback()
+
+def color_diff(diff):
+    '''Colorize lines in the diff generator object'''
+
+    for line in diff:
+        if line.startswith('+'):
+            yield Fore.GREEN + line + Fore.RESET
+        elif line.startswith('-'):
+            yield Fore.RED + line + Fore.RESET
+        else:
+            yield line
+
+class Minion():
+
+    default_config = '/opt/{name}/minion'
+
+    def __init__(self, name=None, config=None):
+        if not config:
+            config = self.default_config.format(name=name)
+        self.config=config
+
+        __opts__ = salt.config.minion_config(self.config)
+        __grains__ = salt.loader.grains(__opts__)
+        __opts__['grains'] = __grains__
+        __utils__ = salt.loader.utils(__opts__)
+        self.salt = salt.loader.minion_mods(__opts__, utils=__utils__)
+
+    @property
+    def highstate(self):
+        return json.loads(json.dumps(self.salt['state.show_highstate'](saltenv='javaci')))
+
+    @property
+    def pillar_items(self):
+        return json.loads(json.dumps(self.salt['pillar.items'](saltenv='javaci')))
+
+
+    @property
+    def grains_items(self):
+        return json.loads(json.dumps(self.salt['grains.items']()))
+
+    def __str__(self):
+        return json.dumps(
+            {
+                "highstate": self.highstate,
+                "pillar_items": self.pillar_items,
+                "grains_items": self.grains_items,
+            },
+            indent=4
+        )
+
+def minions_diff(minionA, minionB):
+    '''Generate multiline string from colorized diffs'''
+
+    comparable_properties = [
+        'highstate',
+        'pillar_items',
+        'grains_items',
+    ]
+
+    diffs = {}
+    for key in comparable_properties:
+        diffs[key] = color_diff (
+            json_delta.udiff(
+                getattr(minionA, key),
+                getattr(minionB, key),
+            )
+        )
+
+    result = "\n".join(
+        [
+            "\n".join( [key.upper(), "\n".join(value)] ) for key, value in diffs.items()
+        ]
+    )
+    return result
+
+
```

Comment:
`if __name__ == '__main__': ...` otherwise this module will execute following lines on import